iFixit/react-commerce
# GA4: Configure to Log And Debug When We Want

## Overview
There are 2 major changes

1. Move the script back into `GoogleAnalytics.tsx` instead of being in `_document.tsx`. This is because we want to access the route for the `debug_mode` state. We don't use "beforeInteractive" to programatically ensure that `window.gtag` exists on pageload now; now we use nextjs Head tag with a script tag which should run before any body code runs.
2.  You can now turn on debug mode by including `ga4_debug=true` in your uri parameters.

## QA

Make sure that everything basically works the same.
Make sure that turning `process.env.NEXT_PUBLIC_GA_DEBUG` stops GA4 events from being logged in debug_mode (you may need someone with vercel access to change this.)
Make sure that adding `ga4_debug=true` as a uri parameter ensures that all GA4 events for that pageload are in debug_mode.

Connects: #1975
